assignment/project/internal/service/tag/service.go

```diff
@@ -64,3 +65,31 @@ func (s *service) List() (*ct.ListTagResponse, error) {
 
 	return prepareListTagResponse(tags), nil
 }
+
+// ListPosts executes all posts retrieval logic by tag id
+func (s *service) ListPosts(id int) (*ct.ListPostResponse, error) {
+	posts, err := s.tagRepo.SelectPost(id)
+	if err != nil {
+		return nil, err
+	}
+
+	users := make([]*model.User, 0, len(posts))
+	tagsLists := make([][]*model.Tag, 0, len(posts))
+	for _, post := range posts {
+		// Get all the tags associated with the post
+		tagsList, err := s.tagRepo.SelectByPost(post.ID)
```

Comment:
ko đặt sql query trong for loop như này nhé
đọc thêm cái này: https://viblo.asia/p/n-1-query-tinh-nang-hay-la-bug-maGK7WnMKj2